RFC8174 Key Word Guidance, Introduction grammatical edits, minor spelling and README fixes.

taylorpaul opened the pull request:
There are miscellaneous minor changes and fixes included in this pull request. I do not believe they substantively change the document. These include the following changes (by commit for cherry-picking) from most to least substantial (not commit order):

- [da38334](https://github.com/ietf-wg-cats/draft-ietf-cats-framework/commit/da38334c0b076df413491e7ddb1f639ab1738f94) Implementing capitalization (or not) of key words that describe specifications or requirements and represent a well-defined meaning via [RFC2119](https://datatracker.ietf.org/doc/rfc2119/) and [RFC8174](https://datatracker.ietf.org/doc/rfc8174/).
- [ac15139](https://github.com/ietf-wg-cats/draft-ietf-cats-framework/commit/ac151393e0faf17c8fd2ff7d85acfc0829c4d7c3) Grammatical edits to the introduction section ONLY to favor active voice or clarify items that confused me on first pass.
- [4b6277c](https://github.com/ietf-wg-cats/draft-ietf-cats-framework/commit/4b6277ca48a5731d04e80d47c7d9ade622f7f2d0) Minor spelling corrections throughout the document.
- [7227c0e](https://github.com/ietf-wg-cats/draft-ietf-cats-framework/commit/7227c0e72b6377f69c980b22f377e212b9fab582) Fix broken CONTRIBUTIONS.md link in README.

muzixing commented on draft-ietf-cats-framework.md:
```diff
@@ -617,7 +617,7 @@ According to the method of distributing and collecting the computing related met
 According to the metric definition in {{?I-D.ietf-cats-metric-definition}}, computing metrics need to be normalized and/or aggregated in order to low down the scalability impact of the existing route system while providing sufficient detail for effective decision-making.
 
-Depeding on the resources and processing capabilities of CATS components, the normalization and aggregation functions can be located in different CATS components. The suggested solution is to implement the normalization and aggregation functions located away from the decision maker, CATS Path Selector (C-PS), especially when C-PS is co-located with CATS-Forwarders. With this in mind, the normaliz ...
+Depending on the resources and processing capabilities of CATS components, the normalization and aggregation functions can be located in different CATS components. The suggested solution is to implement the normalization and aggregation functions located away from the decision maker, CATS Path Selector (C-PS), especially when C-PS is co-located with CATS-Forwarders. With this in mind, the normali ...
```
Thank you for your contributions, really nice.

muzixing commented on draft-ietf-cats-framework.md:
```diff
@@ -131,6 +131,12 @@ Furthermore, this document describes a workflow of the main CATS procedures (see
+"MAY", and "OPTIONAL" in this document are to be interpreted as
+described in BCP 14 {{?RFC2119}} {{?RFC8174}} when, and only when, they
+appear in all capitals, as shown here.
+
```
Good to me

taylorpaul commented on draft-ietf-cats-framework.md:
```diff
@@ -114,19 +114,19 @@ This document describes a framework for Computing-Aware Traffic Steering (CATS).
+Steering in CATS aims to select the appropriate service contact instance to service a request according to a set of network and computing metrics. This selection may not reveal the actual service instance that a client will invoke, e.g., in hierarchical or recursive contexts. Therefore, the metrics of the service contact instance may be  aggregate metrics from multiple service instances.
 
-The CATS framework is an overlay framework for the selection of the suitable service contact instance(s) from a set of candidates. The exact characterization of 'suitable' is determined by a combination of networking and computing metrics.
+The CATS framework is an overlay framework, an abstraction over existing network and compute resources, for the selection of the suitable service contact instance(s) from a set of candidates. A combination of networking and computing metrics determine the exact characterization of services as 'suitable' or not.
```
On my first read it was not clear to me what "overlay framework" meant here.  I noted my confusion in the margins and thus added this line after reading the whole document and understanding the CATS framework better to try and help future readers.

But if it really is obvious to others, happy to drop that part.

taylorpaul replied:
Resolved. Removed the redundant phrase.

taylorpaul commented on draft-ietf-cats-framework.md:
```diff
@@ -140,13 +146,13 @@ Flow:
 Computing-Aware Traffic Steering (CATS):
- :  A traffic engineering approach {{?RFC9522}} that takes into account the dynamic nature of computing resources (e.g., compute and storage) and network state to optimize service-specific traffic forwarding towards a given service contact instance. Various relevant metrics may be used to enable such computing-aware traffic steering policies.
+ :  A traffic engineering approach {{?RFC9522}} that takes into account the dynamic nature of computing resources (e.g., compute and storage) and network state to optimize service-specific traffic forwarding towards a given service contact instance. Various relevant metrics MAY be used to enable such computing-aware traffic steering policies.
 
```
Yeah, I wasn't sure either. The cats-metrics draft does leverage the capitalization but only twice (see MUST usage here: [draft-ietf-cats-metric-definition.md](https://github.com/ietf-wg-cats/draft-ietf-cats-metric-definition/blob/main/draft-ietf-cats-metric-definition.md#design-principles---why-three-metric-levels)).  

I think if we introduce it in the draft-ietf-cats-framework.md we certainly should only capitalize when it meets the RFC2119 definition and we may want to review the use of capitals.

Happy to leave it out, I think consistency is key across the current cats draft.

muzixing replied:
I may suggest to leave it for now, so that we can merge this PR.

taylorpaul commented:
@muzixing I will send a message to the mailing list too, but I believe the pull request is ready as I've done the following:
- Reverted the capitalization commit (if you squash my commits the history of this will get cleaned up)
- Removed the redundant phrase as you suggested.
- Pulled changes from the main repo and rebased off of the last merge request accepted (#125)

Let me know if you need anything else, I want to get this pull-request across the finish line before starting further grammatical edits in other sections.
